src/crypto-providers/ledgerCryptoProvider.ts

```diff
@@ -163,14 +201,30 @@ export const LedgerCryptoProvider: (transport: Transport) => Promise<CryptoProvi
   ): LedgerTypes.TxOutput => {
     const changeAddressParams = getAddressParameters(changeOutputFiles, output.address, network)
     const tokenBundle = output.amount.type === TxTypes.AmountType.WITH_MULTIASSET
-      ? prepareTokenBundle(output.amount.multiasset) : null
-    const datumHashHex = output.datumHash?.toString('hex')
+      ? prepareTokenBundle(output.amount.multiasset) : undefined
+
+    const datumHashHex = prepareDatumHash(output)
+    const datum = prepareDatum(output)
+
+    const scriptHex = output.type === TxTypes.OutputType.MAP_BABBAGE
+      ? output.referenceScript?.toString('hex')
+      : undefined
 
     if (changeAddressParams && areAddressParamsAllowed(signingMode)) {
-      return prepareChangeOutput(output.amount.coin, changeAddressParams, tokenBundle, datumHashHex)
+      return prepareChangeOutput(
+        output.amount.coin,
+        changeAddressParams,
+        tokenBundle,
+        datumHashHex,
+        datum,
+        scriptHex,
+      )
     }
 
     return {
+      type: output.type === TxTypes.OutputType.ARRAY_LEGACY
```

Comment:
Ledgerjs uses `format` instead of `type` now.

Reply:
I temporarily added a .tgz of LedgerJS in https://github.com/vacuumlabs/cardano-hw-cli/pull/127/commits/7064a7993bacb5c334848ed52647ddda755b4888 and change `type` to `format` in https://github.com/vacuumlabs/cardano-hw-cli/pull/127/commits/4172792c61f01ba32272c9e840aa4a773573acf1.

Reply:
I temporarily added interop lib as a .tgz dependency in https://github.com/vacuumlabs/cardano-hw-cli/pull/127/commits/cfb7fa18e2eb2c59981b755dd0ce2b18bdcdbca0.